### DISCUSSION TITLE
calculate topic probabilities after fitting

### DISCUSSION THREAD

**Discussion by aminathshausan** [MaartenGr/BERTopic#774] [Q&A]
Hi, 
I wanted to visualize distribution of topics within a document, using 
`topic_model.visualize_distribution(probabilities[0])` .
However, I missed to state` caculate_probabilities = True` in model fitting (`BERTopic( )`).

Is there a way I can compute the probabilities without re-fitting the model? When I re-fitted the model, it will give different topics due to stochasticity. 

Else, is there a way I could visualise distribution of topics for a given document? 
Thanks

**Comment by vivresursonnuage** [comment]
Hello. I have a similar question: Fitting many data points (approx. 1 M) is no problem. But as soon as soft clustering is applied and the "hdbscan.all_points_membership_vectors" function is called, I run into issues. So the idea was to fit the BERTopic model and transform the dataset in batches and calculate the probabilities afterwards. To conclude, probabilities shouldn't be calculated during the fitting, but later during the transformation. Many thanks in advance for any feedback and assistance.
